pandas/core/indexing.py

```diff
@@ -1814,7 +1814,8 @@ def check_bool_indexer(ax, key):
         result = result.reindex(ax)
         mask = isnull(result._values)
         if mask.any():
-            raise IndexingError('Unalignable boolean Series key provided')
+            raise IndexingError('Unalignable labels in boolean Series index '
+                                '{}'.format(key.index))
```

Comment:
No, sorry, I just meant here in the comments. To see how it looks, as I am not sure if I find it a good idea to include the full index.